crates/l2/proposer/prover_server.rs

```diff
@@ -46,10 +49,141 @@ struct ProverServer {
     on_chain_proposer_address: Address,
     verifier_address: Address,
     verifier_private_key: SecretKey,
+    proving_output_per_block: HashMap<u64, HashMap<ProverType, ProvingOutput>>,
+}
+
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+/// Enum used to identify the different proving systems.
+pub enum ProverType {
+    RISC0,
+    SP1,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Risc0Proof {
+    pub receipt: Box<risc0_zkvm::Receipt>,
+    pub prover_id: Vec<u32>,
+}
+
+pub struct Risc0ContractData {
+    pub block_proof: Vec<u8>,
+    pub image_id: Vec<u8>,
+    pub journal_digest: Vec<u8>,
+}
+
+impl Risc0Proof {
+    pub fn new(receipt: risc0_zkvm::Receipt, prover_id: Vec<u32>) -> Self {
+        Risc0Proof {
+            receipt: Box::new(receipt),
+            prover_id,
+        }
+    }
+
+    pub fn contract_data(&self) -> Result<Risc0ContractData, ProverServerError> {
+        // If we run the prover_client with RISC0_DEV_MODE=0 we will have a groth16 proof
+        // Else, we will have a fake proof.
+        //
+        // The RISC0_DEV_MODE=1 should only be used with DEPLOYER_CONTRACT_VERIFIER=0xAA
+        let block_proof = match self.receipt.inner.groth16() {
+            Ok(inner) => {
+                // The SELECTOR is used to perform an extra check inside the groth16 verifier contract.
+                let mut selector =
+                    hex::encode(inner.verifier_parameters.as_bytes().get(..4).ok_or(
+                        ProverServerError::Custom(
+                            "Failed to get verify_proof_selector in send_proof()".to_owned(),
+                        ),
+                    )?);
+                let seal = hex::encode(inner.clone().seal);
+                selector.push_str(&seal);
+                hex::decode(selector).map_err(|e| {
+                    ProverServerError::Custom(format!("Failed to hex::decode(selector): {e}"))
+                })?
+            }
+            Err(_) => vec![32; 0],
+        };
+
+        let mut image_id: [u32; 8] = [0; 8];
```

Comment:
```suggestion
        let mut image_id = [0u32; 8];
```

Reply:
Done in [c7aa3b7](https://github.com/lambdaclass/ethrex/pull/1466/commits/c7aa3b7938549869411b412f53caf64eac0bbb2e)